unit-testing/presentation.html

```diff
@@ -1,27 +1,310 @@
+        <pre><code data-trim data-noescape>
+            module.exports = {​​
+              roots: ['<rootDir>/src'],​​
+              collectCoverage: true,​​
+              coverageThreshold: {​​ ... },​​
+              preset: 'react-native',​​
+              setupFiles: ['<rootDir>jest.setup.js'],​​
+              setupFilesAfterEnv: ['<rootDir>/src/globalMock.js'],​​
+            };
+          </code></pre>
+      </section>
+      <!-- 16 slide -->
+      <section>
+        <h3 class="slide-name">Structuring Tests​</h3>
+        <pre><code data-trim data-noescape>
+          beforeAll(() => {​
+            // ...​​
+          });​​
+          afterAll(() => {​​
+            // ...​​
+          });​​
+          describe('Our component', () => {​​
+            beforeEach(() => {​​
+             // ...​​
+            });​​
+            it('should test this', () => {​​
+             // ...​​
+            });​​
+          });
+          </code></pre>
+      </section>
+      <!-- 17 slide -->
+      <section>
+        <h3 class="slide-name">Expect​</h3>
+        <pre><code data-trim data-noescape>
+          expect(value)​​
+          expect.anything()​​
+          expect.any(constructor)​​
+          .not​​
+          .toBe(value)​​
+          .toHaveBeenCalled()​​
+          .toBeLessThan(number | bigint)​​
+          .toBeNull()​​
+          .toContain(item)​​
+          .toEqual(value)​​
+          .toMatch(regexp | string)​​
+          .toMatchSnapshot(propertyMatchers?, hint?)​​
+          .toThrow(error?)​​
+        </code></pre>
+      </section>
+      <!-- 18 slide -->
+      <section>
+        <h3 class="slide-name">Mocking & The Jest Object​</h3>
+        <pre><code data-trim data-noescape>
+          // Mock functions​​
+          jest.mock(moduleName, factory)​​
+          jest.unmock(moduleName)​​
+          jest.requireActual(moduleName)​​
+          jest.resetModules()​​
+          jest.isolateModules(fn)​​
+          // Mock modules​​
+          jest.fn(implementation?)​​
+          jest.spyOn(object, methodName)​​
+          jest.clearAllMocks()​​
+          jest.resetAllMocks()​​
+          jest.restoreAllMocks()​​
+        </code></pre>
+      </section>
+      <!-- 19 slide -->
+      <section>
+        <h3 class="slide-name">Snapshots​​</h3>
+        <pre><code data-trim data-noescape>
+          it('should match snapshot', () => {​​
+            const {toJSON} = render(<MyButton title="title" />);​​
+            expect(toJSON()).toMatchSnapshot();​​
+          });​​
+        </code></pre>
+      </section>
+      <!-- 20 slide -->
+      <section>
+        <!-- 20/0 slide -->
+        <section>
+          <h3 class="slide-name">Test coverage​</h3>
+          <p class="font-size-2 align-center">With that being said it is generally accepted that 80% coverage is a good
+            goal to aim for.</p>
+          <p class="font-size-2 align-center">Trying to reach a higher coverage might turn out to be costly, while not
```

Comment:
could you explain in more details it.
probably provide example what should not be covered

Reply:
done